client/my-sites/controller.js

```diff
@@ -187,93 +187,102 @@ module.exports = {
 	siteSelection( context, next ) {
 		const siteID = route.getSiteFragment( context.path );
 		const basePath = route.sectionify( context.path );
-		const currentUser = user.fetching ? null : user.get(); // if we're fetching, should ignore it for now
-		const hasOneSite = currentUser && currentUser.visible_site_count === 1;
 		const allSitesPath = route.sectionify( context.path );
 
-		const redirectToPrimary = () => {
-			let redirectPath = `${ context.pathname }/${ sites.getPrimary().slug }`;
+		const onUserReady = () => {
```

Comment:
@yurynix For my own understanding:

- Why do we need to refetch the user in the case of a domain-only checkout?
- If we need to refetch the user in this case, can do we it in `Checkout`? I'm hesitant to make a bunch of changes to this controller to provide behavior to just one sub-module here.